src/phoenix/trace/trace_dataset.py

```diff
@@ -98,15 +98,19 @@ class TraceDataset:
     """
 
     name: str
+    """
+    A human readable name for the dataset.
+    """
     dataframe: pd.DataFrame
     evaluations: List[Evaluations] = []
-    _id: UUID = uuid4()
+    _id: UUID
     _data_file_name: str = "data.parquet"
 
     def __init__(
         self,
         dataframe: DataFrame,
         name: Optional[str] = None,
+        id: Optional[Union[str, UUID]] = None,
```

Comment:
This doesn't seem like something we should expose on the `__init__`. Why not just initialize inside the `__init__`?

Reply:
I sorta agree but I think if the user wants determinism we let them... It's a bit of a trade-off